src/config/unit_test/mod.rs

```diff
@@ -557,6 +562,31 @@ fn build_input_event(input: &TestInput) -> Result<Event, String> {
             Some(v) => Ok(Event::Log(LogEvent::from_str_legacy(v.clone()))),
             None => Err("input type 'raw' requires the field 'value'".to_string()),
         },
+        "vrl" => {
+            if let Some(source) = &input.source {
+                let fns = vrl::stdlib::all();
+                let result = match vrl::compiler::compile(source, &fns) {
+                    Ok(v) => v,
+                    Err(e) => return Err(Formatter::new(source, e.clone()).to_string()),
+                };
+
+                let mut target = TargetValue {
+                    value: value!({}),
+                    metadata: value::Value::Object(BTreeMap::new()),
+                    secrets: value::Secrets::default(),
+                };
+
+                let mut state = RuntimeState::default();
+                let timezone = TimeZone::default();
+                let mut ctx = Context::new(&mut target, &mut state, &timezone);
+                match result.program.resolve(&mut ctx) {
+                    Ok(value) => Ok(Event::Log(LogEvent::from(value.clone()))),
```

Comment:
It would be nice to expand this to metrics in the future, but this is a good start since unit tests are frequently used for logs. We should make this limitation clear in the docs though.

Reply:
Thats more me being nwe to the code base i suppose! Since the value is just a blob we could probably also parse a metric event from it if the schema matches; i just didnt know how.